SYCL/ESIMD/regression/math_const_fix_test.cpp

```diff
@@ -0,0 +1,235 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==- math_const_fix_test.cpp - Test for tanh -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cmath>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/intel/esimd/simd.hpp>
+#include <sycl/sycl.hpp>
+#include <vector>
+
+constexpr auto sycl_write = sycl::access::mode::write;
+#define SIMD 16
+
+int test_sin_cos_atan(float x) {
+  std::vector<float> vector_result_sin(SIMD);
+  std::vector<float> vector_result_cos(SIMD);
+  std::vector<float> vector_result_atan(SIMD);
+
+  float ha = x;
+  float scalar_result_sin = 0;
+  float scalar_result_cos = 0;
+  float scalar_result_atan = 0;
+
+  {
+    sycl::queue queue;
+
+    sycl::buffer<float, 1> vector_buffer(vector_result_sin.data(),

```

Comment:
I suggest not to use buffers/accessors in new tests because these will not run on newer archs.